src/streaming-main.cpp

```diff
@@ -0,0 +1,659 @@
+    Core::ArgsParser args_parser;
+    args_parser.set_general_help("LibJS test262 runner for streaming tests");
+    args_parser.add_option(s_harness_file_directory, "Directory containing the harness files", "harness-location", 'h', "harness-files");
+    args_parser.add_option(s_use_bytecode, "Use the bytecode interpreter", "use-bytecode", 'b');
+    args_parser.add_option(s_parse_only, "Only parse the files", "parse-only", 'p');
+    args_parser.add_option(timeout, "Seconds before test should timeout", "timeout", 't', "seconds");
+    args_parser.parse(argc, argv);
+
+    if (s_harness_file_directory.is_empty()) {
+        dbgln("You must specify the harness file directory");
+        return 2;
+    }
+
+    if (!s_harness_file_directory.ends_with('/')) {
+        s_harness_file_directory = String::formatted("{}/", s_harness_file_directory);
+    }
+
+    if (timeout <= 0) {
+        dbgln("timeout must be atleast 1");
+        return 2;
+    }
+
+    // The piping stuff is based on https://stackoverflow.com/a/956269.
+    constexpr auto BUFFER_SIZE = 1 * KiB;
+    char buffer[BUFFER_SIZE] = {};
+
+    auto saved_stdout = dup(STDOUT_FILENO);
+    if (saved_stdout < 0) {
+        perror("dup");
+        return 1;
+    }
+
+    saved_stdout_fd = fdopen(saved_stdout, "w");
+    if (!saved_stdout_fd) {
+        perror("fdopen");
+        return 1;
+    }
+
+    int stdout_pipe[2];
+    if (pipe(stdout_pipe) < 0) {
+        perror("pipe");
+        return 1;
+    }
+
+    auto flags = fcntl(stdout_pipe[0], F_GETFL);
+    flags |= O_NONBLOCK;
+    fcntl(stdout_pipe[0], F_SETFL, flags);
+
+    if (dup2(stdout_pipe[1], STDOUT_FILENO) < 0) {
+        perror("dup2");
+        return 1;
+    }
+
+    if (close(stdout_pipe[1]) < 0) {
+        perror("close");
+        return 1;
+    }
+
+    auto collect_output = [&] {
+        fflush(stdout);
+        auto nread = read(stdout_pipe[0], buffer, BUFFER_SIZE);
+        String value;
+
+        if (nread > 0) {
+            value = String { buffer, static_cast<size_t>(nread) };
+            while (nread > 0) {
+                nread = read(stdout_pipe[0], buffer, BUFFER_SIZE);
+            }
+        }
+
+        return value;
+    };
+
+    auto stdin = Core::File::standard_input();
+
+    if (signal(SIGVTALRM, timer_handler) == SIG_ERR) {
+        perror("signal");
+        return 1;
+    }
+
+    struct itimerval ten_seconds;
+    ten_seconds.it_value.tv_sec = timeout;
+    ten_seconds.it_value.tv_usec = 0;
+    ten_seconds.it_interval.tv_sec = 0;
+    ten_seconds.it_interval.tv_usec = 0;
+
+    struct itimerval disarm;
+    disarm.it_value.tv_sec = 0;
+    disarm.it_value.tv_usec = 0;
+    disarm.it_interval.tv_sec = 0;
+    disarm.it_interval.tv_usec = 0;
+
+#define ARM_TIMER()                                       \
+    if (setitimer(ITIMER_VIRTUAL, &ten_seconds, 0) < 0) { \
+        perror("setitimer");                              \
+        return 1;                                         \
+    }
```

Comment:
I read that `setitimer` might not be portable. Is this a problem? 
Linux and mac seem to have it. But I have no idea if this can be a problem

Reply:
Dr. POSIX has deprecated these in favour of using `timer_settime`:
https://pubs.opengroup.org/onlinepubs/9699919799/functions/getitimer.html